pkg/dnsserver/dnscontroller/handler.go

```diff
@@ -0,0 +1,43 @@
+package dnscontroller
```

Comment:
`dnscontroller` package doesn't really apply here. How about `server` so the Handler is referenced as `server.Handler`?